includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -2912,6 +2944,30 @@ static function ( $selector_language ) {
 		}
 	}
 
+	/**
+	 * Given list of class names, create a regular expression pattern to match them in a selector.
+	 *
+	 * @since 1.4
+	 *
+	 * @param string[] $class_names Class names.
+	 * @return string Pattern,
+	 */
+	private static function get_class_name_selector_pattern( $class_names ) {
+		$pattern  = '/\.(';
+		$pattern .= implode(
+			'|',
+			array_map(
+				function ( $class_name ) {
+					return preg_quote( $class_name, '/' );
+				},
+				$class_names
```

Comment:
We have no guarantee here we'll receive an array, so we should be defensive:
```suggestion
				(array) $class_names
```